evap/middleware.py

```diff
@@ -19,7 +21,10 @@ def process_view(request, view_func, _view_args, _view_kwargs):
         if "no_login_required" in view_func.__dict__ and view_func.no_login_required:
             return None
 
-        if view_func.__name__ in ["OIDCAuthenticationRequestView", "OIDCAuthenticationCallbackView"]:
+        if view_func.__name__ in _ALLOWED_WITHOUT_LOGIN:
+            return None
```

Comment:
Thinking about this again: We have a test for this. The behavior of django and mozilla-django-oidc should be deterministic with a fixed version (either view_func.__name__ or view_func.view_class.__name__ should be set). Since the test always runs against the version in requirements.txt, and the code is only ever supposed to run with these versions, I think it's safe to remove the old `if` here.

And if it's safe, we should do it (we don't want to untested dead code lying around)

Reply:
This was an exception for oidc from the beginning, so we can only do the necessary check (as long as we also run the tests on production before deploying :D). I would give the constant a less general name (or inline again).

Reply:
I don't think we've ever run the tests on production. So I guess it's time to play the debug game and find out what happened here (tried to circumvent that):

In 3.2.13, django still populated the generated view-method with a `__name__` [here](https://github.com/django/django/blob/3.2.13/django/views/generic/base.py#L74).
Somewhere in between (probably on 4 update), django started intentionally leaving `__name__` unpopulated, arguing that `view_class` should be used. The `view_class` attribute [is documented](https://docs.djangoproject.com/en/3.2/ref/class-based-views/base/#django.views.generic.base.View.as_view), and should have already worked with earlier versions.

So, seems that `view_func.view_class.__name__` should work according to documentation as long as mozilla-django-oidc doesn't add breaking changes (changing from view class to view func, or changing their names).

I'll drop the "old" check, as I think this likely not going to break (ha, I jinxed it)